scripts/OverlaySurvey.py

```diff
@@ -0,0 +1,194 @@
+            for out_peers in topology[key]["outboundPeers"]:
+                results.append(out_peers["nodeId"])
+    return results
+
+def send_requests(peer_list):
+    for key in peer_list:
+        PARAMS["node"] = key
+        requests.get(url = SURVEY_REQUEST, params = PARAMS) 
+
+def check_results():
+    r = requests.get(url = SURVEY_RESULT)
+    data = r.json()
+
+    if("topology" not in data):
+        return []
+
+    topology = data["topology"]
+
+    for key in topology:
+        if topology[key] is None:
+            continue
+
+        if "inboundPeers" in topology[key] and topology[key]["inboundPeers"]:
+            for in_peers in topology[key]["inboundPeers"]:
+                merged_results[key]["inboundPeers"][in_peers["nodeId"]] = in_peers
+                add_new_node(key)
+                add_new_node(in_peers["nodeId"])
+                add_new_edge(in_peers["nodeId"], key)
+            merged_results[key]["totalInbound"] = topology[key]["numTotalInboundPeers"]
+        if "outboundPeers" in topology[key] and topology[key]["outboundPeers"]:
+            for out_peers in topology[key]["outboundPeers"]:
+                merged_results[key]["outboundPeers"][out_peers["nodeId"]] = out_peers
+                add_new_node(key)
+                add_new_node(out_peers["nodeId"])
+                add_new_edge(key, out_peers["nodeId"])
+            merged_results[key]["totalOutbound"] = topology[key]["numTotalOutboundPeers"]
+    
+        merged_results[key]["responseSeen"] = True
+
+    return get_next_peers(topology)
+
+def write_graph_stats():
+    stats = {}
+    stats["average_shortest_path_length"] = nx.average_shortest_path_length(G)
+    stats["average_clustering"] = nx.average_clustering(G)
+    stats["clustering"] = nx.clustering(G)
+    stats["degree"] = dict(nx.degree(G))
+    with open('graphStats.json', 'w') as outfile:
+        json.dump(stats, outfile)
+
+#not used yet
+def get_node_aliases():
+    r = requests.get(url = "https://api.stellarbeat.io/v1/nodes")
+    nodeList = r.json()
+    
+    result = {}
+    for node in nodeList:
+        if "alias" in node:
+            result[node["publicKey"]] = node["alias"]
+    return result
+    
+
+# construct the argument parse and parse the arguments
+ap = argparse.ArgumentParser()
+ap.add_argument("-n", "--node", help="address of initial survey node")
+ap.add_argument("-d", "--duration", help="duration of survey in seconds")
+ap.add_argument("-nl", "--nodeList", help="optional list of seed nodes")
+ap.add_argument("-gml", "--graphml", help="print stats for the graphml input graph")
+
+try:
+    args = ap.parse_args()
+except:
+    print("Error with input arguments!")
+    ap.print_help()
+    sys.exit(0)
+
+#we just want to print stats for an existing graph
+if args.graphml:
+    G = nx.read_graphml(args.graphml)
+    write_graph_stats()
+    sys.exit(0)
+
+if not args.node or not args.duration:
```

Comment:
You can mark the args as required in argparse, don't need to double check here.

Reply:
Issue with that is these variables are not required if you're just using the -gml flag (analyzing an existing graphml file)

Reply:
Looks like the right way to do this is with subparsers

Reply:
fixed